dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java

```diff
@@ -191,28 +194,32 @@ public WebMessage setSystemSettingV29(@RequestBody Map<String, Object> settings)
   // -------------------------------------------------------------------------
 
   @GetMapping(value = "/{key}", produces = ContextUtils.CONTENT_TYPE_TEXT)
-  public @ResponseBody Serializable getSystemSettingOrTranslationAsPlainText(
+  public @ResponseBody ResponseEntity<String> getSystemSettingOrTranslationAsPlainText(
       @PathVariable("key") String key,
       @RequestParam(value = "locale", required = false) String locale,
       HttpServletResponse response,
       @CurrentUser User currentUser) {
+    if (!settingExistsAndIsNotConfidential(key)) {
+      return ResponseEntity.status(404).body(SETTING_DOESNT_EXIST_OR_CONFIDENTIAL);
```

Comment:
Could we also use the `NotFoundException` instead to make this more visible for OpenAPI? Then you should also be able to use `String` as return type.

Reply:
I did try that, to try to keep them all the same but the server bombs out with a 500. This was the only way I could get a 404 with an error message back to the client, which keeps the behaviour as close as possible.